src/client.ts

```diff
@@ -223,14 +224,17 @@ export class OpenFeatureClient implements Client {
 
       return evaluationDetails;
     } catch (err: unknown) {
-      const errorCode = (!!err && (err as { code: string }).code) || GENERAL_ERROR;
+
+      const errorMessage: string = (err as Error)?.message;
+      const errorCode: ErrorCode = (err as OpenFeatureError)?.code || ErrorCode.GENERAL;
```

Comment:
Only behavioral change. Get code from `.code` or default to GENERAL, get message from `.message`.